model/project_ref.go

```diff
@@ -1489,3 +1440,92 @@ func GetUpstreamProjectName(triggerID, triggerType string) (string, error) {
 	}
 	return upstreamProject.DisplayName, nil
 }
+
+func pipelineForValueIsBool(projectKey, repoKey string, val bool) []bson.M {
+	return []bson.M{
+		lookupRepoStep,
+		{"$match": bson.M{
+			"$or": []bson.M{
+				{projectKey: val},
+				{projectKey: bson.M{"$ne": false}, bsonutil.GetDottedKeyName("repo_ref", repoKey): val},
```

Comment:
Is the intent of this query to return whether or not a specific setting is considered true/false for a branch? If so shouldn't this part be "projectKey is null or does not exist"?